lxd/instance/instance_utils.go

```diff
@@ -693,6 +697,23 @@ func CreateInternal(s *state.State, args db.InstanceArgs, clearLogDir bool) (Ins
 
 	args.Config["volatile.uuid.generation"] = args.Config["volatile.uuid"]
 
+	cpuPinningSpecified := strings.ContainsAny(args.Config["limits.cpu"], ",-")
+
+	if args.Type == instancetype.VM {
```

Comment:
We dont want to populate the defaults into the instance DB record, this isn't how LXD's config defaults work.

Instead you should consult the instance's expanded config where you would use the `limits.cpu` value and consider then whether to use it or not based on the value of `limits.cpu.pin_strategy`.